src/firecfg/main.c

```diff
@@ -487,5 +486,9 @@ int main(int argc, char **argv) {
 	if (arg_debug)
 		printf("%s %d %d %d %d\n", user, getuid(), getgid(), geteuid(), getegid());
 
+	// if runs as regular user, fix .desktop files in ~/.local/share/applications directory
+	if (getuid() != 0)
```

Comment:
If we check `geteuid`, it would continue to work with sudo.

Reply:
I don't see why, since `firecfg` doesn't have the `setuid` bit.

Also, `sudo` sets both `uid` and `euid` to the same value.
You can check it by running `sudo firecfg --debug`.

Reply:
Your right so far, but sudo also sets the `SUDO_USER` env-var, which is checked by `get_user` 

https://github.com/netblue30/firejail/blob/2359fa901d6ac4a266ca2a1c0ddf31939a3a9863/src/firecfg/main.c#L286-L287

which is used here
https://github.com/netblue30/firejail/blob/2359fa901d6ac4a266ca2a1c0ddf31939a3a9863/src/firecfg/main.c#L323-L328

and then here

https://github.com/netblue30/firejail/blob/2359fa901d6ac4a266ca2a1c0ddf31939a3a9863/src/firecfg/main.c#L476-L484

Reply:
Agreed. But I don't see what it has to do with `geteuid()`.

`sudo firecfg` is still working as expected on my system.

Reply:
If you start it with sudo, the real user-id is always 0, right? And `getuid` checks the real-user-id (https://linux.die.net/man/2/getuid). Therefore fix_desktop_files is never called. When checking the effective-uid it would get called.

what am I missing?

Reply:
`sudo` sets both `uid` and `euid` to the same value. So, if you start it with `sudo`, both the real user ID and the effective user ID will be set to `0`.

I think you are missing the permissions drop that happens just before:
https://github.com/netblue30/firejail/blob/2359fa901d6ac4a266ca2a1c0ddf31939a3a9863/src/firecfg/main.c#L482-L483

After these lines, both `uid` and `euid` are set back to the original user ID.

Reply:
https://linux.die.net/man/2/setuid
If I had read not only the first, but also the second sentence…